packages/core/src/components/icon/icon.tsx

```diff
@@ -17,50 +17,40 @@ export interface IIconProps extends IIntentProps, IProps {
      * Name of the icon (with or without `"pt-icon-"` prefix).
      * If `undefined`, this component will render nothing.
      */
-    iconName: IconName | undefined;
+    iconName: LegacyIconName | undefined;
 
     /**
-     * Size of the icon.
-     * Blueprint provides each icon in two sizes: 16px and 20px. The keyword `"inherit"` will
-     * render a 20px icon but inherit `font-size` from its parent.
-     * Constants are exposed for each of these values on the component itself:
-     * `Icon.SIZE_(STANDARD|LARGE|INHERIT)`,
+     * Height of icon. A number value will be interpreted as pixels. Use a string value for other units.
+     * By default, inherits height from surrounding styles, such as `line-height`.
+     * @default "inherit"
+     */
+    height?: number | string;
+
+    /**
+     * Width of icon. A number value will be interpreted as pixels. Use a string value for other units.
      * @default 16
      */
-    iconSize?: 16 | 20 | "inherit";
+    width?: number | string;
 }
 
 export class Icon extends React.PureComponent<IIconProps & React.HTMLAttributes<HTMLSpanElement>, never> {
     public static displayName = "Blueprint.Icon";
 
     public static readonly SIZE_STANDARD = 16 as 16;
     public static readonly SIZE_LARGE = 20 as 20;
-    public static readonly SIZE_INHERIT = "inherit" as "inherit";
 
     public render() {
-        if (this.props.iconName == null) {
+        const { className, iconName, intent, width = 16, height = "inherit" } = this.props;
+        if (iconName == null) {
             return null;
         }
-        const { className, iconName, intent, iconSize = Icon.SIZE_STANDARD, ...restProps } = this.props;
-
-        const classes = classNames(
-            getSizeClass(iconSize),
-            Classes.iconClass(iconName),
-            Classes.intentClass(intent),
-            className,
+        const shortName = iconName.replace("pt-icon-", "") as IconName;
+        const classes = classNames(Classes.ICON, Classes.iconClass(iconName), Classes.intentClass(intent), className);
+        return (
+            <svg className={classes} width={width} height={height} viewBox="0 0 16 16">
+                <title>{iconName}</title>
+                {IconSvgPaths[shortName].map((d, i) => <path key={i} d={d} clip-rule="evenodd" fill-rule="evenodd" />)}
```

Comment:
I was imagining that the consumer would feed this `IconSvgPaths` as a prop (`iconSet`?), so they can use this component with custom icon sets. Would also solve the 16px / 20px switch, since these two sizes are actually two different sets

Reply:
oh heck no, we want this to be as easy as possible for _blueprint_ users. i think this `Icon` component should be the Blueprint icon component. if you want to use your own icons, you should do `icon={<MyOwnIcon />}` instead of `icon="time"`.

Reply:
Oh I see, that wasn't clear to me, name feels very generic. Then maybe what you want is something like `<Icon16 />` `<Icon20 />` to reflect the icon set?

Reply:
What if the default behavior would choose 16 or 20 based on the given `iconSize`? like, the one that requires less scaling? just gotta figure out the math.

`Dialog` is also a generic name 😄